subfun/pmCon.m

```diff
@@ -30,109 +30,107 @@
     % check all the steps specified in the model
     for iStep = 1:length(model.Steps)
 
-        % only compute contrasts at the subject level
-        if strcmp(model.Steps{iStep}.Level, 'subject')
-
-            % specify all the contrasts
-
-            % first the contrasts to compute automatically against baseline
-            for iCon = 1:length(model.Steps{iStep}.AutoContrasts)
-
-                con_counter = con_counter + 1;
-
-                C = zeros(1,size(SPM.xX.X,2));
-
-                % get condition name
-                cdt_name = model.Steps{iStep}.AutoContrasts{iCon};
-                cdt_name = strrep(cdt_name, 'trial_type.', '');
-
-                % get regressors index corresponding to the HRF of that condition
-                regIdx = strfind(SPM.xX.name', [' ' cdt_name '*bf(1)']);
-                regIdx = ~cellfun('isempty', regIdx);  %#ok<*STRCL1>
-
-                % give them a value of 1
-                C(end,regIdx) = 1;
-
-                % stores the specification
-                contrasts(con_counter).C = C;
-                contrasts(con_counter).name =  cdt_name;
-
-            end
-
-            % then the contrasts that involve contrasting conditions
-            % amongst themselves or soemthing inferior to baseline
-            for iCon = 1:length(model.Steps{iStep}.Contrasts)
-
-                con_counter = con_counter + 1;
-
-                C = zeros(1,size(SPM.xX.X,2));
-
-                for iCdt = 1:length(model.Steps{iStep}.Contrasts(iCon).ConditionList)
-
-                    % get condition name
-                    cdt_name = model.Steps{iStep}.Contrasts(iCon).ConditionList{iCdt};
-                    cdt_name = strrep(cdt_name, 'trial_type.', '');
-
+        switch model.Steps{iStep}.Level
+            
+            %% compute contrasts at the subject level
+            case 'subject'
+                
+                % specify all the contrasts
+                
+                % first the contrasts to compute automatically against baseline
+                for iCon = 1:length(model.Steps{iStep}.AutoContrasts)
+                    
+                    con_counter = con_counter + 1;
+                    
+                    C = zeros(1,size(SPM.xX.X,2));
+                    
```

Comment:
Seems git diff is getting quite confused here. Might be better checking the whole file.